server/commitlog/message.go

```diff
@@ -17,6 +17,7 @@ type Message struct {
 	Key        []byte
 	Value      []byte
 	Headers    map[string][]byte
+	Offset     int64
```

Comment:
This should go under the transient fields below since this value will not get serialized, e.g. in the case of OCC being disabled.

Reply:
I was thinking so as well. 
WIP

Reply:
Taking a look into that, I realize that it is actually always initialized to 0.

in `partition.natsToProtoMessage()`

```golang
	if message != nil {
		m.Key = message.Key
		m.Value = message.Value
		for key, value := range message.Headers {
			m.Headers[key] = value
		}
		m.AckInbox = message.AckInbox
		m.CorrelationID = message.CorrelationId
		m.AckPolicy = message.AckPolicy
		m.Offset = message.Offset
	}
```

If OCC is disabled for the stream (default behavior), `m.Offset` is set to 0 will not change anything 

If OCC is enabled for the stream , this value set to 0 may potentially gives error, as the signil value is -1. I guess we expect the client who wants to use this new feature must take care of the fact that default Offset value would be -1 to preserve current behavior.

So I think it's same to assume that this value is also serialized.

Is that the case or do I miss something ?

Reply:
The offset that is serialized is that which is assigned by the commit log. E.g. if OCC is disabled but for some reason the client sets the expected `offset` anyway, this value is discarded, which is why I think it's a "transient" field, i.e. it's only passed in to do the OCC check.